src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -14,6 +14,8 @@ namespace Microsoft.AspNetCore.Components.Forms;
 /// </summary>
 public abstract class InputBase<TValue> : ComponentBase, IDisposable
 {
+    private static readonly ExpressionFormatter s_expressionFormatter = new();
```

Comment:
Yeah, I wasn't sure if we were going to do something like define some `IInputNameFormatter` service interface, where only some platforms would provide an implementation. But if this is something we will enable for any supported platform (which seems fine to me, I think), then it could definitely be static.

Reply:
We could do a check before calling it inside `ValueExpressionAsString` as an option. Most likely, it'll be a property we set on the edit context